### ISSUE TITLE
[TypeScript] Request syntax diagnostics when "typescript.tsserver.useSeparateSyntaxServer": "forAllRequests"

### ISSUE THREAD

**Issue by amcasey** [microsoft/vscode#127700]
After a week or so of using partial mode full-time, my biggest complaint is that it doesn't flag obvious syntactic errors - particularly unbalanced braces.  I can see why we'd want to avoid having partial mode show squiggles when it's enabled temporarily, during project load, but I think it might be nice to enable syntactic diagnostics when `forAllRequests` is set.  Having said that, we almost certainly don't want semantic diagnostics, because there will be tons of unresolved symbols and `any`s. 

This will probably require enabling a capability (`semantic`?) that allows `getErr` requests to be sent and, unless it is gated on a correspondingly updated TS version, a filter to drop `semanticDiag` and `suggestionDiag` events.

**Comment by amcasey** [comment]
FYI @mjbvz

**Comment by mjbvz** [comment]
Fixed on the VS Code side by 8b3d6668bdc3ecd9c2c692a34d6dd92624c3efed

Closing now that upstream fix has been merged too

**Comment by mjbvz** [comment]
To verify:

1. In our repo, `cd extensions` then `yarn add typescript@next` (we still ship TS 4.3 by default since 4.4 with the fix is not yet stable)
2. `yarn compile-web`
3. `yarn web` to start web
4. In a TS file, try adding some syntax errors and confirm these are reported as errors

**Comment by alexr00** [comment]
Following those instructions got me this error:
"Activating extension 'vscode.typescript-language-features' failed: Invalid destructuring assignment target."

**Comment by mjbvz** [comment]
@amcasey I think something may have changed/regressed on the TypeScript side. I see `geterr` requests in web but don't see any events being generated

To test this out:

1. Checkout `dev/mjbvz/bundle-44`
2. `yarn`
3. `yarn watch` and then also `yarn compile-web`
4. Start browser with `yarn web`

Here's an example `geterr` request:

```
[Trace  - 05:13:34.887] <syntax> Sending request: geterr (191). Response expected: yes. Current queue length: 0
Arguments: {
    "delay": 0,
    "files": [
        "^memfs:/sample-folder/large.ts"
    ]
}
```

**Comment by amcasey** [comment]
@mjbvz Looking

**Comment by amcasey** [comment]
I'm seeing 
```
Err 20    [13:43:25.417] Exception on executing command delayed processing of request 7:

    Illegal invocation

    TypeError: Illegal invocation
        at t.delay (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3148256)
        at s.g.updateErrorCheck (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3176374)
        at s.g.getDiagnostics (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3201463)
        at http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3163087
        at t.executeAction (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3148701)
        at t.startNew (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3147490)
        at handlers.e.Map.e.getEntries.n.<computed> (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3163056)
        at http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3216726
        at s.g.executeWithRequestId (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3216550)
        at s.g.executeCommand (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3216680)
        at s.g.onMessage (http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3217600)
        at http://localhost:8080/static/extensions/typescript-language-features/dist/browser/typescript/tsserver.web.js:15:3280217

Info 21   [13:43:25.417] Session does not support events: ignored event: {"seq":0,"type":"event","event":"requestCompleted","body":{"request_seq":7}}
```

**Comment by amcasey** [comment]
@mjbvz It works for me in `code.bat`, but not in `yarn web`.  I had only ever validated the functionality in `code.bat` - had you seen it in the web version?

**Comment by mjbvz** [comment]
Yes I believe it used to work in the web as well

Is that error coming TS server? We did update from webpack 4 -> 5 since I tested this as well so could bundling be causing any issues?

**Comment by amcasey** [comment]
Yeah, it's from tsserver.web.js.  The F12 tools keep freezing when I try to break into it - possibly because there are 3M characters on that line.

**Comment by amcasey** [comment]
I've created https://github.com/microsoft/TypeScript/pull/45242 to fix the exception, but the web server will still [drop](https://github.com/microsoft/TypeScript/blob/db0f7938ddd6da30a3160867ed08a916f4d2f965/src/webServer/webServer.ts#L197) all events on the floor (not new).  Can you please either file a feature request against TS or skip the call to `getErr` in this scenario?

**Comment by mjbvz** [comment]
Great find @amcasey!

I swear I saw this working but I think I was using a custom branch of TypeScript. We need to use `getErr` to get any diagnostics, correct?

**Comment by amcasey** [comment]
Most of the credit goes to @andrewbranch. :smile:

Yes, you would need to use `getErr` for any diagnostics.  If you really wanted to, I suppose you could request diagnostics synchronously (i.e. without events), but that might fail for other reasons.

**Comment by mjbvz** [comment]
Ah I thought https://github.com/microsoft/TypeScript/pull/44859 also would cover web? Was that only for desktop partial mode? If so, I'll open a new issue for supporting syntax errors on web too

**Comment by amcasey** [comment]
It does cover web in the sense that the event is generated.  Unfortunately, it turns out that web then drops that event on the floor, rather than sending it back to the editor.  I agree that the scenario seems important, so please go ahead a file a new issue.

**Comment by DanielRosenwasser** [comment]
What was the reasoning behind not supporting events in the first place? Just not needing them?

**Comment by amcasey** [comment]
@DanielRosenwasser I misread - it does drop the event on the floor, but that's because `!canUseEvents` and not because it's a web server.

**Comment by mjbvz** [comment]
Upstream issue with fix for web https://github.com/microsoft/TypeScript/pull/45313

**Comment by mjbvz** [comment]
Fixed by pulling in latest TS version

For verification, you should just be able to use `yarn web` now and check that introducing syntax errors in to a TS file generates an error
